pywhy_graphs/algorithms/generic.py

```diff
@@ -855,3 +856,99 @@ def all_vstructures(G: nx.DiGraph, as_edges: bool = False):
                 else:
                     vstructs.add((p1, node, p2))  # type: ignore
     return vstructs
+
+
+def check_back_arrow(G: ADMG, X, Y: set):
```

Comment:
Let's make this a private function?

Also, document that this checks if any node in `Y` points back to `X`, where `X` is a node, not a set? Or did I misunderstand that?

Reply:
Was about to do it. I just wanted to know your opinion on how to test it, the PR is not ready for a full review yet. I should have been more explicit. 😅

Reply:
> Also, document that this checks if any node in `Y` points back to `X`, where `X` is a node, not a set? Or did I misunderstand that?

Yes you understood that correctly, it is never called for a set X.